File: DataOutput/clsOutputFileHandles.vb
Namespace DataOutput
    Public Class clsOutputFileHandles
        Inherits clsMasicEventNotifier

        Public Property ScanStats As StreamWriter
        Public Property SICDataFile As StreamWriter
        Public Property XMLFileForSICs As Xml.XmlTextWriter
        Public Property MSMethodFilePathBase As String
        Public Property MSTuneFilePathBase As String

        Public Sub CloseScanStats()
            If Not ScanStats Is Nothing Then
                ScanStats.Close()
                ScanStats = Nothing
            End If
        End Sub

        Public Function CloseAll() As Boolean

            Try
                CloseScanStats()

                If Not SICDataFile Is Nothing Then
                    SICDataFile.Close()
                    SICDataFile = Nothing
                End If

                If Not XMLFileForSICs Is Nothing Then
                    XMLFileForSICs.Close()
                    XMLFileForSICs = Nothing
                End If

                Return True
            Catch ex As Exception
                ReportError("Error in CloseOutputFileHandles", ex)
                Return False
            End Try
        End Function
    End Class

End Namespace

File: DataOutput/clsOutputFileHandles.cs
using System;
using System.IO;

namespace MASIC.DataOutput
{
    /// <summary>
    /// Container for output file handles
    /// </summary>
    public class clsOutputFileHandles : clsMasicEventNotifier
    {
        /// <summary>
        /// Scan stats file
        /// </summary>
        public StreamWriter ScanStats { get; set; }

        // ReSharper disable CommentTypo

        /// <summary>
        /// SIC details file
        /// </summary>
        /// <remarks>This is different than _SICstats.txt file</remarks>
        public StreamWriter SICDataFile { get; set; }

        // ReSharper restore CommentTypo

        /// <summary>
        /// XML Results file
        /// </summary>
        public System.Xml.XmlTextWriter XMLFileForSICs { get; set; }

        /// <summary>
        /// MS method file base path
        /// </summary>
        public string MSMethodFilePathBase { get; set; }

        /// <summary>
        /// MS tune file base path
        /// </summary>
        public string MSTuneFilePathBase { get; set; }

        /// <summary>
        /// Close the _ScanStats.txt file
        /// </summary>
        public void CloseScanStats()
        {
            if (ScanStats != null)
            {
                ScanStats.Close();
                ScanStats = null;
            }
        }

        /// <summary>
        /// Close all files
        /// </summary>
        public bool CloseAll()
        {
            try
            {
                CloseScanStats();
                if (SICDataFile != null)
                {
                    SICDataFile.Close();
                    SICDataFile = null;
                }

                if (XMLFileForSICs != null)
                {
                    XMLFileForSICs.Close();
                    XMLFileForSICs = null;
                }

                return true;
            }
            catch (Exception ex)
            {
                ReportError("Error in CloseOutputFileHandles", ex);
                return false;
            }
        }
    }
}
